# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieService.cpp
# Commit: b2c86eba88a3
# Full Hash: b2c86eba88a348fb99ed75b0c1aed59848c0d451
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-09-17 17:41 +0000
# Regressor Bug: 1982261
# File Overlap Count: 1
# Overlapping Files: netwerk/cookie/CookieService.cpp
# Description:
#   Bug 1982261 - CookieServiceParent logs about invalid cookies instead of crashing,  a=dmeehan DONTBUILD
# ==============================================================================

diff -r b889a3daa95f -r b2c86eba88a3 netwerk/cookie/CookieService.cpp
--- a/netwerk/cookie/CookieService.cpp	Wed Sep 17 13:33:37 2025 +0000
+++ b/netwerk/cookie/CookieService.cpp	Wed Sep 17 17:41:13 2025 +0000
@@ -1700,15 +1700,13 @@
   return mPersistentStorage;
 }
 
-bool CookieService::SetCookiesFromIPC(const nsACString& aBaseDomain,
-                                      const OriginAttributes& aAttrs,
-                                      nsIURI* aHostURI, bool aFromHttp,
-                                      bool aIsThirdParty,
-                                      const nsTArray<CookieStruct>& aCookies,
-                                      BrowsingContext* aBrowsingContext) {
+nsICookieValidation::ValidationError CookieService::SetCookiesFromIPC(
+    const nsACString& aBaseDomain, const OriginAttributes& aAttrs,
+    nsIURI* aHostURI, bool aFromHttp, bool aIsThirdParty,
+    const nsTArray<CookieStruct>& aCookies, BrowsingContext* aBrowsingContext) {
   if (!IsInitialized()) {
     // If we are probably shutting down, we can ignore this cookie.
-    return true;
+    return nsICookieValidation::eOK;
   }
 
   CookieStorage* storage = PickStorage(aAttrs);
@@ -1720,7 +1718,7 @@
     MOZ_ASSERT(validation);
 
     if (validation->Result() != nsICookieValidation::eOK) {
-      return false;
+      return validation->Result();
     }
 
     // create a new Cookie and copy attributes
@@ -1738,7 +1736,7 @@
                        aBrowsingContext);
   }
 
-  return true;
+  return nsICookieValidation::eOK;
 }
 
 void CookieService::GetCookiesFromHost(